@here/harp-mapview/lib/AnimatedExtrusionHandler.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { TileKey } from "@here/harp-geoutils";
+import { DataSource } from "./DataSource";
+import { MapView } from "./MapView";
+import { Tile } from "./Tile";
+
+/**
+ * Animation states for extrusion effect
+ */
+export enum AnimatedExtrusionState {
+    None,
+    Started,
+    Playing,
+    Finished
+}
+
+/**
+ * Creates `AnimatedExtrusionHandler` that controls animated extrusion effect for
+ * extruded objects in [[Tile]] when they appear on the scene.
+ */
+export class AnimatedExtrusionHandler {
+    /**
+     * Animate the extrusion of the buildings if set to `true`.
+     */
+    static DEFAULT_ENABLED = true;
+    /**
+     * Duration of the building's extrusion in milliseconds
+     */
+    static DEFAULT_DURATION = 400;
+    /**
+     * Start animation when defined zoom level exceeded
+     */
+    static DEFAULT_ZOOM_LEVEL = 16;
+    /**
+     * Minimum ratio value for extrusion effect
+     */
+    static DEFAULT_RATIO_MIN: number = 0.001;
+    /**
+     * Maximum ratio value for extrusion effect
+     */
+    static DEFAULT_RATIO_MAX: number = 1;
+
+    private m_zoomLevel: number;
+    private m_zoomLevelPrevious: number;
+    private m_tilesSet: Set<Tile> = new Set();
+    private m_tileKeysAnimatedSet: Set<TileKey> = new Set();
+
+    /**
+     * Constructs a `MapViAnimatedExtrusionHandlerewFog` instance.
+     *
+     * @param m_mapView Instance of [[MapView]] that passes `zoomLevel`
+     * through the `zoom` poperty update
+     */
+    constructor(private m_mapView: MapView) {
+        this.m_zoomLevelPrevious = this.m_mapView.zoomLevel;
+        this.m_zoomLevel = this.m_mapView.zoomLevel;
+    }
+
+    /**
+     * [[MapView]] updates zoom level. Then calculate actual extrusion ratio and trigger animation
+     */
+    set zoom(zoomLevel: number) {
+        // if zoomLevel has been changed since last render
+        if (this.m_zoomLevel !== zoomLevel) {
```

Comment:
If you use `this.m_zoomLevel` like this, then `...Previous` is not needed.

Actually, `m_zoomLevel` is not needed ans thou shall use only `m_zoomLevelPrevious`